config/config.go

```diff
@@ -263,6 +258,105 @@ func FromViper(v *viper.Viper) (*Config, error) {
 	return cfg, nil
 }
 
+func determineTreeRoot(v *viper.Viper, cfg *Config) error {
+	var err error
+
+	// enforce the various tree root options are mutually exclusive
+	// some of this is being done for us at the flag level, but you can also set these values in config or environment
+	// variables.
+	count := 0
+
+	if cfg.TreeRoot != "" {
+		count++
+	}
+
+	if cfg.TreeRootCmd != "" {
+		count++
+	}
+
+	if cfg.TreeRootFile != "" {
+		count++
+	}
+
+	if count > 1 {
+		return errors.New("only one of tree-root, tree-root-cmd or tree-root-file can be specified")
+	}
+
+	// set git-based tree root command if the walker is git and no tree root has been specified
+	if cfg.Walk == walk.Git.String() && count == 0 {
+		cfg.TreeRootCmd = "git rev-parse --show-toplevel"
+
+		log.Infof(
+			"git walker enabled, but tree root has not been specified: defaulting tree-root-cmd to '%s'",
```

Comment:
nit, feel free to ignore: The tone of "but" sounds to me like something unexpected or surprising has happened here, and I don't think that's the case.

```suggestion
			"git walker enabled, and tree root has not been specified: defaulting tree-root-cmd to '%s'",
```

Reply:
We can also drop the comma if switching to "and":

```suggestion
			"git walker enabled and tree root has not been specified: defaulting tree-root-cmd to '%s'",
```